.github/workflows/pr-close-delete-env.yml

```diff
@@ -5,25 +5,61 @@ on:
     types: [closed]
   workflow_dispatch:
     inputs:
-      environment:
-        description: "Pull request number"
+      slotID:
+        description: "Pull request number (Add multiple IDs seperated by a space)"
         required: true
         default: ""
+  workflow_call:
+    inputs:
+      slotIDs:
+        type: string
+        description: "Slot ID (Add multiple IDs seperated by a space)"
+        required: true
 
 defaults:
   run:
     shell: pwsh
 
 env:
-  SLOT_NAME: pr-${{ github.event.number }}
-
+  SLOT_NAME: ${{ inputs.slotIDs || inputs.slotID || github.event.number }}
+  SLOT_PREFIX: pr-
 permissions:
   id-token: write
   contents: read
 
 jobs:
+  setting-up-slot-ids:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Converting slots into JSON Formatted Array
+        id: set-matrix
+        run: |
+          # When we have a list of IDs
+          if('${{ env.SLOT_NAME }}'.Contains(' ')){
+            $slotsArray = '${{ env.SLOT_NAME }}' -split ' '
+            $slotsArrayInJson = $slotsArray | ConvertTo-Json -Compress
+            echo "matrix=$slotsArrayInJson" >> $env:GITHUB_OUTPUT
+          } else {
+            # When we have a single slot ID,
+            # Adding extra space at the tail to split it into an array
+
+            $slotsArray = '${{ env.SLOT_NAME }} ' -split ' '
+            $slotsArrayInJson = $slotsArray | ConvertTo-Json -Compress
+
+            # Removing the extra space now e.g [slotID,""]
+
+            $filteredSlotID = $slotsArrayInJson.Replace(',""', '')
+            echo "matrix=$filteredSlotID" >> $env:GITHUB_OUTPUT
+          }
```

Comment:
I would expect that this step isn't needed if the inputs were json compatible

Reply:
We are handling 3 different triggers, therefore, I added this step to filter out the inputs and then convert them accordingly. Initially, I was passing the JSON formatted input from another flow, but for the `workflow dispatch` and `close` triggers, I had to convert them anyway.

Reply:
if you do the above changes, this step will not be required